### PR TITLE
[FIXED JENKINS-34881] - handle non-default security settings for new installs

### PR DESCRIPTION
**Pull request by kzantow**
Some applications may provide an init script or otherwise configure security before the setup wizard runs. This correctly handles these situations by avoiding presenting the unlock screen and avoiding the 'create user' step in the setup wizard.

This addresses: https://issues.jenkins-ci.org/browse/JENKINS-34881

@reviewbybees esp. @oleg-nenashev

### CODE REVIEW

File: core/src/main/java/jenkins/install/InstallUtil.java [hunk 1]
```diff
@@ -193,7 +188,7 @@ private static InstallState getDefaultInstallState() {
             // Edge case: used Jenkins 1 but did not save the system config page,
             // the version is not persisted and returns 1.0, so try to check if
             // they actually did anything
-            if (!j.getItemMap().isEmpty() || !mayBeJenkins2SecurityDefaults(j) || !j.getNodes().isEmpty()) {
```
Comment (kzantow) [inline core/src/main/java/jenkins/install/InstallUtil.java]: We cannot rely on security settings for determining if this is an upgrade or new install; scripts may set up security

**Comment by stephenc** [comment]
:bee: AIUI (but if the security settings are present do we force a login to admin role before starting the wizard? (Remember the security realm could be something like Google apps authentication so presenting a username/password form would be a spectacular fail)

**Comment by kzantow** [comment]
@stephenc we use the default behavior of the security realm, e.g. if you set LegacySecurity realm, with anyone can do anything in an init script, it would bypass login altogether. Maybe I need some guidance what should happen here?

**Comment by stephenc** [comment]
*before* you start the wizard you need to either:

* have the default settings in play; or
* be logged in as an administrator

So basically the page that starts the flow needs to require admin permissions if not using the new admin lockdown mode.

I think it is ok to block at that login screen before continuing if the wizard is going to be displayed.

**Comment by kzantow** [comment]
@stephenc thanks for the feedback, it was already redirecting to the admin page, but it seemed unreliable; adding the extra `checkPermission` here seems to do the trick in both cases

**Comment by oleg-nenashev** [comment]
@kzantow the behavior is strange. At least my SSO engine does not work well
1) I get the admin lockdown screen with TMP password on startup
2) When I enter the TMP password and submit, I get the login error 
3) After that...
* If I click on the "login" hyperlink, I get the user/password dialog. SSO does not work
* When I refresh the page, SSO works well. Jenkins displays the plugin selection screen

🐜 , seems redirect to an errorLogin by permission check is a wrong approach. Maybe page refresh? It should require login for the current permission check from what I see

**Comment by oleg-nenashev** [comment]
@kzantow gentle bump

**Comment by kzantow** [comment]
@oleg-nenashev I tracked down the issue that the token-based login page was not forwarding the 'from' parameter; please have a look at this again

**Comment by oleg-nenashev** [comment]
The change looks good. Testing it

**Comment by oleg-nenashev** [comment]
@kzantow 
the last change didn't help at east in my case :(

**Comment by oleg-nenashev** [comment]
Retested the change again with jenkins-2.7. Everything works fine. Maybe it was a glitch caused by other fixes OR maybe I've just built the stuff improperly.

Works like a charm for me 🐝

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Comment by oleg-nenashev** [comment]
I am about merging this PR on the evening if there is no extra feedback

**Pull request merged**